pkg/controller/spoketoken/spoke_token_controller.go

```diff
@@ -244,27 +254,27 @@ func (r *ReconcileAgentToken) prepareAgentTokenSecret(token string) *corev1.Secr
 	return mcSecret
 }
 
-func (r *ReconcileAgentToken) getServiceAccountTokenSecret(sa corev1.ServiceAccount) *corev1.Secret {
-	// Get the service account token from the service account's secret list
-	saSecret := &corev1.Secret{}
-
-	for _, secret := range sa.Secrets {
-		secretName := types.NamespacedName{
-			Name:      secret.Name,
-			Namespace: sa.Namespace,
-		}
+func (r *ReconcileAgentToken) getServiceAccountTokenSecret() *corev1.Secret {
+	// Get all secrets
+	// list thing for rolling update check
+	secretList := &corev1.SecretList{}
+	listopts := &client.ListOptions{Namespace: "open-cluster-management-agent-addon"}
+	err := r.Client.List(context.TODO(), secretList, listopts)
 
-		if err := r.Client.Get(context.TODO(), secretName, saSecret); err != nil {
-			continue
-		}
+	if err != nil {
+		klog.Error(err.Error())
+		return nil
+	}
 
-		// Get the service account token
-		if saSecret.Type == corev1.SecretTypeServiceAccountToken {
-			break
+	for _, secret := range secretList.Items {
+		// Get the application-manager service account token
+		if secret.Type == corev1.SecretTypeServiceAccountToken && strings.HasPrefix(secret.Name, "application-manager-token-") {
```

Comment:
Is it possible to run into potential conflict? Discussing in https://github.com/stolostron/multicloud-operators-subscription/pull/788#discussion_r920653895